salt/_modules/metalk8s_kubernetes_cronjob.py

```diff
@@ -95,14 +106,23 @@ def _set_cronjob_suspend(name, namespace, suspend, **kwargs):
         True: if the suspend state has been updated.
     """
     cronjob = get_cronjob(name, namespace, **kwargs)
+    patch = {
+        "spec": {"suspend": suspend},
+        "metadata": {"annotations": cronjob["metadata"].get("annotations", {})},
+    }
+
+    if suspend and mark is not None:
+        patch["metadata"]["annotations"]["metalk8s.scality.com/suspend_mark"] = mark
+    else:
+        patch["metadata"]["annotations"]["metalk8s.scality.com/suspend_mark"] = None
```

Comment:
You do not need to have all annotations it's a patch so it merges things

```suggestion
    patch = {
        "spec": {"suspend": suspend},
        "metadata": {"annotations": {}),
    }

    if suspend and mark is not None:
        patch["metadata"]["annotations"]["metalk8s.scality.com/suspend_mark"] = mark
    else:
        patch["metadata"]["annotations"]["metalk8s.scality.com/suspend_mark"] = None
```

Reply:
Done.